src/vs/workbench/api/node/mainThreadHandlerRegistry.ts

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+/**
+ * MainThreadHandlerRegistry maintains a collection of objects by handle.
+ */
```

Comment:
Okay, I just didn't understand why we need a "handler registry" for the progress callbacks. In my head passing progress callbacks doesn't involve such complex structures?